### PR TITLE
[FPGA][NFC] improves tests for [[intel::no_global_work_offset]] attribute

### PR DESCRIPTION
**Pull request by smanna12**
This patch updates test bases for [[intel::no_global_work_offset]] attribute:
1. separates AST and diagnostics to make it easier to read/follow-up.
2. removes duplicate test cases.
3. no compiler changes.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/test/SemaSYCL/intel-fpga-no-global-work-offset-ast.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+// CHECK-NEXT: value: Int 1
+// CHECK-NEXT: SubstNonTypeTemplateParmExpr
+// CHECK-NEXT: NonTypeTemplateParmDecl
+// CHECK-NEXT: IntegerLiteral {{.*}} 'int' 1
+
+// Test that checks template parameter suppport on function.
+template <int N>
+[[intel::no_global_work_offset(N)]] void func1() {}
+
+// Test that checks template parameter suppport on member function of class template.
```
Comment (elizabethandrews) [inline clang/test/SemaSYCL/intel-fpga-no-global-work-offset-ast.cpp]: support*

File: clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp [hunk 2]
```diff
@@ -1,68 +1,61 @@
...
-        []() [[intel::no_global_work_offset("foo")]]{});
+// Diagnostic is emitted because the arguments mismatch.
+[[intel::no_global_work_offset(0)]] void func6(); // expected-note {{previous attribute is here}}
+[[intel::no_global_work_offset(1)]] void func6(); // expected-warning{{attribute 'no_global_work_offset' is already applied with different arguments}}
 
-    h.single_task<class test_kernel7>([]() {
-      // expected-error@+1{{'no_global_work_offset' attribute only applies to functions}}
-      [[intel::no_global_work_offset(1)]] int a;
-    });
+// Test that checks template parameter suppport on function.
```
Comment (elizabethandrews) [inline clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp:52]: support*

File: clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp [hunk 3]
```diff
@@ -1,68 +1,61 @@
...
-    // expected-error@+2{{integral constant expression must have integral or unscoped enumeration type, not 'const char[4]'}}
-    h.single_task<class test_kernel6>(
-        []() [[intel::no_global_work_offset("foo")]]{});
+// Diagnostic is emitted because the arguments mismatch.
+[[intel::no_global_work_offset(0)]] void func6(); // expected-note {{previous attribute is here}}
+[[intel::no_global_work_offset(1)]] void func6(); // expected-warning{{attribute 'no_global_work_offset' is already applied with different arguments}}
 
-    h.single_task<class test_kernel7>([]() {
-      // expected-error@+1{{'no_global_work_offset' attribute only applies to functions}}
-      [[intel::no_global_work_offset(1)]] int a;
```
Comment (elizabethandrews) [inline clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp:60]: Did we lose coverage for this?

Comment (smanna12) [reply clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp:60]: No, I moved this to [clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp](https://github.com/intel/llvm/pull/6184/files#diff-0d420fe07fdb14fef73cb83c5ac0962db1d4e2dc5fca293220852776999d2fdd)

Comment (elizabethandrews) [reply clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp:60]: Ah ok. I didn't see it when I looked earlier. Thanks!

**Review by elizabethandrews — APPROVED** [review]

File: clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp (hunk 3 above)
Comment (smanna12) [reply clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp:60]: No problem. Thanks for reviews!

**Comment by smanna12** [comment]
Thank you for reviews.

@intel/llvm-gatekeepers, this PR is ready to be merged. Thank you

**Pull request merged**